### PR TITLE
#159324289 - Built out diary entries API endpoints 

### PR DESCRIPTION
**Pull request by iampinheirosamuel**
**What does this PR do?**

It has CRUD on resources endpoint of MyDiary APIs

**Description of Tasks to be completed**

It has the following endpoint working

``POST /api/v1/entries``

``GET /api/v1/entries``

``GET /api/v1/entries/:entry_id``

``PUT /api/v1/entries/:entry_id``

``DELETE /api.v1/entries/:entry_id``

**How should it be tested manually**

After cloning the repo, cd into the roof folder and run ``npm start``

-Using Postman test every endpoint above with this header:

``key``: Content-Type
``value``: application/json

**What are the relevant pivotal tracker stories**
#159324289  #159298218 #159299445 #159299319 #159299544 #159324289

### CODE REVIEW

File: controller/controller.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import data from '../models/model';
+
+class diary {
+  static getEntries(req, res) {
```
Comment (fob413) [inline controller/controller.js:4]: Explain how this works

File: controller/controller.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    const today = new Date();
+    const entry = {};
+    if (typeof req.body.name === 'string' && typeof req.body.content === 'string') {
+      entry.title = req.body.name;
+      entry.content = req.body.content;
+      entry.created_at = today.toLocaleDateString('en-US', { year: 'numeric', month: 'long', day: 'numeric' });
+      data.push(entry);
+      return res.status(201).json({
+        RESPONSE: 'entry added successfully',
+        SUCCESS: { entry },
```
Comment (fob413) [inline controller/controller.js]: update you response to this format, across all controllers
```
res.status().json({
    status: 'success/fail',
    message: 'whatever message you want your user to get',
    data: {}
})
```

- Update your tests accordingly

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          res.body.should.be.a('object');
+          res.body.should.have.property('SUCCESS');
+          res.body.SUCCESS.should.be.a('object');
+          done();
+        });
+    });
+  });
+});
+
+// describe('MyDiary Users', () => {
```
Comment (fob413) [inline test/test.js]: Please remove all commented out code if they're not in use

File: controller/controller.js (hunk 1 above)
Comment (iampinheirosamuel) [reply controller/controller.js:4]: The controller handle the logic behind the CRUD operation of user's requests before persisting data to the database(data structure in this case).  I created a class diary with static methods each handling user's requests.

-getEntries method handles the GET all entries request from the user. Once the request is made the method returns a response based on the entries stored.

Comment (iampinheirosamuel) [reply controller/controller.js:4]: The controller handles the CRUD operations of user's request. It is made up of a class 'diary' with static methods handling each request of the user. I imported the model containing an array named 'data'. The exchange of data is done through the static methods in the controller class.

-getEntries method GET user's request and return response of all entries to the user stored within the data structure(array).

-getEntry method GET user's request, validates it and return response of an entry to the user stored within the data structure(array). If request is invalid, it returns a response with 'Not Found' message

-postEntry method handles POST user request, validates it and add an entry to the database with a response sent to the user if valid. If user request not valid, return with a response to user of 'entry not added'.

-modifyEntry method handles PUT user request validates it and return response of the entry modified  within the data structure(array). If request is invalid, it returns a response with 'Not Found' message

-deleteEntry method handles DELETE user request validates it and return response with message of entry deleted from the data structure(array). If request is invalid, it returns a response with 'Not Found' message

**Review by fob413 — APPROVED** [review]

**Review by fob413 — APPROVED** [review]
Go ahead and merge

**Pull request merged**